<!-- Phidgets GPS launch file -->

<launch>

  <!-- PARAMETERS -->
  <arg name="application_namespace" default="fuseon"/>
  <arg name="device_namespace" default="phidgets_gps"/>
  <arg name="node_name" default="phidgets_gps_sync"/>

  <arg name="frame_id" default="phidgets_gps"/>

  <arg name="position_stdev" default="0.02"/>

  <arg name="serial_number" default="286007"/>

  <arg name="nav_sat_fix_topic_name" default="gps/nav_sat_fix"/>
  <arg name="nav_sat_status_topic_name" default="gps/nav_sat_status"/>
  <arg name="time_reference_topic_name" default="time_reference"/>
  

  <!-- NODE -->
  <group ns="$(arg application_namespace)/$(arg device_namespace)">
    <node name="phidgets_gps_sync_node" pkg="phidgets_gps" type="phidgets_gps_sync_node" output="screen">

      <!--  -->
      <param name="frame_id" type="string" value="$(arg frame_id)"/>

      <!--  -->
      <param name="position_stdev" type="double" value="$(arg position_stdev)"/>

      <!-- SERIAL NUMBER -->
      <param name="serial_number" type="int" value="$(arg serial_number)"/>

      <!-- TOPIC NAMES -->
      <param name="nav_sat_fix_topic_name" type="string" value="$(arg nav_sat_fix_topic_name)"/>
      <param name="nav_sat_status_topic_name" type="string" value="$(arg nav_sat_status_topic_name)"/>
      <param name="time_reference_topic_name" type="string" value="$(arg time_reference_topic_name)"/>

    </node>
  </group>


</launch>
